artichoke-backend/src/extn/core/kernel/mod.rs

```diff
@@ -90,7 +90,7 @@ impl Kernel {
         let interp = unwrap_interpreter!(mrb);
 
         for value in args.iter() {
-            let s = Value::new(&interp, *value).to_s();
+            let s = String::from_utf8_lossy(&Value::new(&interp, *value).to_s()).to_string();
```

Comment:
let's add a followup to also have `State::print` and `State::puts` take `Vec<u8>` as well. Thanks for putting in this work to make that next step more obvious :)

Reply:
See GH-415.